src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryDate.cs

```diff
@@ -402,10 +402,41 @@ private static FormulaValue DateParse(EvalVisitor runner, EvalVisitorContext con
             }
         }
 
+        private static bool CultureExists(string name)
+        {
+            CultureInfo[] availableCultures =
+                CultureInfo.GetCultures(CultureTypes.AllCultures);
+
+            foreach (CultureInfo culture in availableCultures)
+            {
+                if (string.Equals(culture.Name, name, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
         public static FormulaValue DateTimeParse(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, StringValue[] args)
         {
             var str = args[0].Value;
-            if (DateTime.TryParse(str, runner.CultureInfo, DateTimeStyles.None, out var result))
+
+            // argCI will have Cultural info in-case one was passed in argument else it will have the default one.
+            CultureInfo culture = runner.CultureInfo;
+            if (args.Length > 1)
+            {
+                if (CultureExists(args[1].Value))
```

Comment:
>  if (CultureExists(args[1].Value)) [](http://example.com/codeflow?start=14&length=35)

the pattern we use:
bool TryGetCulture(name, out value)

So this should be:
if (!TryGetCulture(args[1].Value, out culture))
{
    return // error.
}

Reply:
Fixed this!